From e254c1bb4780d27e48a6445ceacab135c057eafb Mon Sep 17 00:00:00 2001
From: Aihua Xu <aihuaxu@gmail.com>
Date: Wed, 27 May 2015 09:26:00 -0700
Subject: [PATCH 0264/1164] HIVE-10826 : Support min()/max() functions over x
 preceding and y preceding windowing (Aihua Xu via
 Ashutosh Chauhan)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>

Conflicts:
	ql/src/test/queries/clientpositive/windowing_windowspec2.q
	ql/src/test/results/clientpositive/windowing_windowspec2.q.out
---
 .../hadoop/hive/ql/udf/generic/GenericUDAFMax.java |   45 +-
 .../hadoop/hive/ql/udf/generic/GenericUDAFMin.java |    8 +-
 .../queries/clientpositive/windowing_windowspec2.q |   17 +
 .../clientpositive/windowing_windowspec2.q.out     | 1302 ++++++++++++++++++++
 4 files changed, 1350 insertions(+), 22 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMax.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMax.java
index 6b7808a..33600f2 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMax.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMax.java
@@ -130,9 +130,7 @@ public Object terminate(AggregationBuffer agg) throws HiveException {
 
     @Override
     public GenericUDAFEvaluator getWindowingEvaluator(WindowFrameDef wFrmDef) {
-      BoundaryDef start = wFrmDef.getStart();
-      BoundaryDef end = wFrmDef.getEnd();
-      return new MaxStreamingFixedWindow(this, start.getAmt(), end.getAmt());
+      return new MaxStreamingFixedWindow(this, wFrmDef);
     }
 
   }
@@ -168,7 +166,7 @@ public GenericUDAFEvaluator getWindowingEvaluator(WindowFrameDef wFrmDef) {
 
       public State(AggregationBuffer buf) {
         super(buf);
-        maxChain = new ArrayDeque<Object[]>(numPreceding + numFollowing + 1);
+        maxChain = new ArrayDeque<Object[]>(wFrameDef.isStartUnbounded() ? 1 : wFrameDef.getWindowSize());
       }
 
       @Override
@@ -180,7 +178,7 @@ public int estimate() {
         if (underlying == -1) {
           return -1;
         }
-        if (numPreceding == BoundarySpec.UNBOUNDED_AMOUNT) {
+        if (wFrameDef.isStartUnbounded()) {
           return -1;
         }
         /*
@@ -189,7 +187,7 @@ public int estimate() {
          * underlying * wdwSz sz of maxChain = sz of underlying * wdwSz
          */
 
-        int wdwSz = numPreceding + numFollowing + 1;
+        int wdwSz = wFrameDef.getWindowSize();
         return underlying + (underlying * wdwSz) + (underlying * wdwSz)
             + (3 * JavaDataModel.PRIMITIVES1);
       }
@@ -202,8 +200,8 @@ protected void reset() {
     }
 
     public MaxStreamingFixedWindow(GenericUDAFEvaluator wrappedEval,
-        int numPreceding, int numFollowing) {
-      super(wrappedEval, numPreceding, numFollowing);
+        WindowFrameDef wFrmDef) {
+      super(wrappedEval, wFrmDef);
     }
 
     @Override
@@ -235,26 +233,32 @@ public void iterate(AggregationBuffer agg, Object[] parameters)
         }
       }
 
+      // We need to insert 'null' before processing first row for the case: X preceding and y preceding
+      if (s.numRows == 0) {
+        for (int i = wFrameDef.getEnd().getRelativeOffset(); i < 0; i++) {
+          s.results.add(null);
+        }
+      }
+
       /*
        * add row to chain. except in case of UNB preceding: - only 1 max needs
        * to be tracked. - current max will never become out of range. It can
        * only be replaced by a larger max.
        */
-      if (numPreceding != BoundarySpec.UNBOUNDED_AMOUNT
-          || s.maxChain.isEmpty()) {
+      if (!wFrameDef.isStartUnbounded() || s.maxChain.isEmpty()) {
         o = o == null ? null : ObjectInspectorUtils.copyToStandardObject(o,
             inputOI(), ObjectInspectorCopyOption.JAVA);
         s.maxChain.addLast(new Object[] { o, s.numRows });
       }
 
-      if (s.numRows >= numFollowing) {
+      if (s.numRows >= wFrameDef.getEnd().getRelativeOffset()) {
         s.results.add(s.maxChain.getFirst()[0]);
       }
       s.numRows++;
 
       int fIdx = (Integer) s.maxChain.getFirst()[1];
-      if (numPreceding != BoundarySpec.UNBOUNDED_AMOUNT
-          && s.numRows > fIdx + numPreceding + numFollowing) {
+      if (!wFrameDef.isStartUnbounded()
+          && s.numRows >= fIdx +  wFrameDef.getWindowSize()) {
         s.maxChain.removeFirst();
       }
     }
@@ -277,19 +281,26 @@ private boolean isGreater(Object in, Object last) {
     public Object terminate(AggregationBuffer agg) throws HiveException {
 
       State s = (State) agg;
-      Object[] r = s.maxChain.getFirst();
+      Object[] r = s.maxChain.isEmpty() ? null : s.maxChain.getFirst();
 
-      for (int i = 0; i < numFollowing; i++) {
+      // After all the rows are processed, continue to generate results for the rows that results haven't generated.
+      // For the case: X following and Y following, process first Y-X results and then insert X nulls.
+      // For the case X preceding and Y following, process Y results.
+      for (int i = Math.max(0, wFrameDef.getStart().getRelativeOffset()); i < wFrameDef.getEnd().getRelativeOffset(); i++) {
         s.results.add(r[0]);
         s.numRows++;
         int fIdx = (Integer) r[1];
-        if (numPreceding != BoundarySpec.UNBOUNDED_AMOUNT
-            && s.numRows - numFollowing + i > fIdx + numPreceding
+        if (!wFrameDef.isStartUnbounded()
+            && s.numRows + i >= fIdx + wFrameDef.getWindowSize()
             && !s.maxChain.isEmpty()) {
           s.maxChain.removeFirst();
           r = !s.maxChain.isEmpty() ? s.maxChain.getFirst() : r;
         }
       }
+      for (int i = 0; i < wFrameDef.getStart().getRelativeOffset(); i++) {
+        s.results.add(null);
+        s.numRows++;
+      }
 
       return null;
     }
diff --git a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMin.java b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMin.java
index d931d52..816350f 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMin.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDAFMin.java
@@ -125,9 +125,7 @@ public Object terminate(AggregationBuffer agg) throws HiveException {
 
     @Override
     public GenericUDAFEvaluator getWindowingEvaluator(WindowFrameDef wFrmDef) {
-      BoundaryDef start = wFrmDef.getStart();
-      BoundaryDef end = wFrmDef.getEnd();
-      return new MinStreamingFixedWindow(this, start.getAmt(), end.getAmt());
+      return new MinStreamingFixedWindow(this, wFrmDef);
     }
 
   }
@@ -135,8 +133,8 @@ public GenericUDAFEvaluator getWindowingEvaluator(WindowFrameDef wFrmDef) {
   static class MinStreamingFixedWindow extends MaxStreamingFixedWindow {
 
     public MinStreamingFixedWindow(GenericUDAFEvaluator wrappedEval,
-        int numPreceding, int numFollowing) {
-      super(wrappedEval, numPreceding, numFollowing);
+        WindowFrameDef wFrmDef) {
+      super(wrappedEval, wFrmDef);
     }
 
     protected ObjectInspector inputOI() {
diff --git a/ql/src/test/queries/clientpositive/windowing_windowspec2.q b/ql/src/test/queries/clientpositive/windowing_windowspec2.q
index 3e8aa93..991f256 100644
--- a/ql/src/test/queries/clientpositive/windowing_windowspec2.q
+++ b/ql/src/test/queries/clientpositive/windowing_windowspec2.q
@@ -22,3 +22,20 @@ select ts, f, sum(f) over (partition by ts order by f rows between unbounded pre
 select ts, f, sum(f) over (partition by ts order by f rows between 1 following and 2 following) from over10k limit 100;
 select ts, f, sum(f) over (partition by ts order by f rows between unbounded preceding and 1 following) from over10k limit 100;
 
+-- count
+select ts, f, count(f) over (partition by ts order by f rows between 2 preceding and 1 preceding) from over10k limit 100;
+select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 preceding) from over10k limit 100;
+select ts, f, count(f) over (partition by ts order by f rows between 1 following and 2 following) from over10k limit 100;
+select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 following) from over10k limit 100;
+
+-- max
+select ts, f, max(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100;
+select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100;
+select ts, f, max(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100;
+select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100;
+
+-- min
+select ts, f, min(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100;
+select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100;
+select ts, f, min(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100;
+select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100;
diff --git a/ql/src/test/results/clientpositive/windowing_windowspec2.q.out b/ql/src/test/results/clientpositive/windowing_windowspec2.q.out
index 0879344..2775445 100644
--- a/ql/src/test/results/clientpositive/windowing_windowspec2.q.out
+++ b/ql/src/test/results/clientpositive/windowing_windowspec2.q.out
@@ -476,3 +476,1305 @@ POSTHOOK: Input: default@over10k
 2013-03-01 09:11:58.703072	71.68	802.1099938452244
 2013-03-01 09:11:58.703072	79.46	882.1299904882908
 2013-03-01 09:11:58.703072	80.02	963.4199914038181
+PREHOOK: query: -- count
+select ts, f, count(f) over (partition by ts order by f rows between 2 preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: -- count
+select ts, f, count(f) over (partition by ts order by f rows between 2 preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	3.17	0
+2013-03-01 09:11:58.70307	10.89	1
+2013-03-01 09:11:58.70307	14.54	2
+2013-03-01 09:11:58.70307	14.78	2
+2013-03-01 09:11:58.70307	17.85	2
+2013-03-01 09:11:58.70307	20.61	2
+2013-03-01 09:11:58.70307	28.69	2
+2013-03-01 09:11:58.70307	29.22	2
+2013-03-01 09:11:58.70307	31.17	2
+2013-03-01 09:11:58.70307	38.35	2
+2013-03-01 09:11:58.70307	38.61	2
+2013-03-01 09:11:58.70307	39.48	2
+2013-03-01 09:11:58.70307	40.54	2
+2013-03-01 09:11:58.70307	41.6	2
+2013-03-01 09:11:58.70307	46.08	2
+2013-03-01 09:11:58.70307	54.36	2
+2013-03-01 09:11:58.70307	56.94	2
+2013-03-01 09:11:58.70307	64.96	2
+2013-03-01 09:11:58.70307	73.52	2
+2013-03-01 09:11:58.70307	78.58	2
+2013-03-01 09:11:58.70307	81.41	2
+2013-03-01 09:11:58.70307	84.71	2
+2013-03-01 09:11:58.70307	87.43	2
+2013-03-01 09:11:58.70307	91.36	2
+2013-03-01 09:11:58.70307	92.96	2
+2013-03-01 09:11:58.70307	95.04	2
+2013-03-01 09:11:58.703071	0.83	0
+2013-03-01 09:11:58.703071	1.99	1
+2013-03-01 09:11:58.703071	3.73	2
+2013-03-01 09:11:58.703071	8.86	2
+2013-03-01 09:11:58.703071	10.62	2
+2013-03-01 09:11:58.703071	11.32	2
+2013-03-01 09:11:58.703071	12.83	2
+2013-03-01 09:11:58.703071	14.7	2
+2013-03-01 09:11:58.703071	14.96	2
+2013-03-01 09:11:58.703071	17.58	2
+2013-03-01 09:11:58.703071	19.1	2
+2013-03-01 09:11:58.703071	21.01	2
+2013-03-01 09:11:58.703071	26.95	2
+2013-03-01 09:11:58.703071	27.23	2
+2013-03-01 09:11:58.703071	29.07	2
+2013-03-01 09:11:58.703071	29.71	2
+2013-03-01 09:11:58.703071	31.84	2
+2013-03-01 09:11:58.703071	31.94	2
+2013-03-01 09:11:58.703071	35.32	2
+2013-03-01 09:11:58.703071	37.32	2
+2013-03-01 09:11:58.703071	38.5	2
+2013-03-01 09:11:58.703071	42.08	2
+2013-03-01 09:11:58.703071	44.3	2
+2013-03-01 09:11:58.703071	44.66	2
+2013-03-01 09:11:58.703071	46.84	2
+2013-03-01 09:11:58.703071	48.89	2
+2013-03-01 09:11:58.703071	49.64	2
+2013-03-01 09:11:58.703071	50.28	2
+2013-03-01 09:11:58.703071	52.09	2
+2013-03-01 09:11:58.703071	53.26	2
+2013-03-01 09:11:58.703071	54.09	2
+2013-03-01 09:11:58.703071	56.45	2
+2013-03-01 09:11:58.703071	56.76	2
+2013-03-01 09:11:58.703071	61.41	2
+2013-03-01 09:11:58.703071	61.88	2
+2013-03-01 09:11:58.703071	63.03	2
+2013-03-01 09:11:58.703071	64.55	2
+2013-03-01 09:11:58.703071	68.62	2
+2013-03-01 09:11:58.703071	76.13	2
+2013-03-01 09:11:58.703071	79.05	2
+2013-03-01 09:11:58.703071	80.43	2
+2013-03-01 09:11:58.703071	81.41	2
+2013-03-01 09:11:58.703071	82.85	2
+2013-03-01 09:11:58.703071	83.98	2
+2013-03-01 09:11:58.703071	84.21	2
+2013-03-01 09:11:58.703071	85.55	2
+2013-03-01 09:11:58.703071	87.93	2
+2013-03-01 09:11:58.703071	88.93	2
+2013-03-01 09:11:58.703071	94.27	2
+2013-03-01 09:11:58.703071	99.45	2
+2013-03-01 09:11:58.703072	0.36	0
+2013-03-01 09:11:58.703072	0.48	1
+2013-03-01 09:11:58.703072	0.79	2
+2013-03-01 09:11:58.703072	1.27	2
+2013-03-01 09:11:58.703072	4.48	2
+2013-03-01 09:11:58.703072	9.0	2
+2013-03-01 09:11:58.703072	23.27	2
+2013-03-01 09:11:58.703072	25.13	2
+2013-03-01 09:11:58.703072	25.34	2
+2013-03-01 09:11:58.703072	25.91	2
+2013-03-01 09:11:58.703072	29.01	2
+2013-03-01 09:11:58.703072	30.47	2
+2013-03-01 09:11:58.703072	37.95	2
+2013-03-01 09:11:58.703072	39.3	2
+2013-03-01 09:11:58.703072	45.91	2
+2013-03-01 09:11:58.703072	52.44	2
+2013-03-01 09:11:58.703072	54.1	2
+2013-03-01 09:11:58.703072	56.7	2
+2013-03-01 09:11:58.703072	58.77	2
+2013-03-01 09:11:58.703072	62.09	2
+2013-03-01 09:11:58.703072	68.2	2
+2013-03-01 09:11:58.703072	71.68	2
+2013-03-01 09:11:58.703072	79.46	2
+2013-03-01 09:11:58.703072	80.02	2
+PREHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	3.17	0
+2013-03-01 09:11:58.70307	10.89	1
+2013-03-01 09:11:58.70307	14.54	2
+2013-03-01 09:11:58.70307	14.78	3
+2013-03-01 09:11:58.70307	17.85	4
+2013-03-01 09:11:58.70307	20.61	5
+2013-03-01 09:11:58.70307	28.69	6
+2013-03-01 09:11:58.70307	29.22	7
+2013-03-01 09:11:58.70307	31.17	8
+2013-03-01 09:11:58.70307	38.35	9
+2013-03-01 09:11:58.70307	38.61	10
+2013-03-01 09:11:58.70307	39.48	11
+2013-03-01 09:11:58.70307	40.54	12
+2013-03-01 09:11:58.70307	41.6	13
+2013-03-01 09:11:58.70307	46.08	14
+2013-03-01 09:11:58.70307	54.36	15
+2013-03-01 09:11:58.70307	56.94	16
+2013-03-01 09:11:58.70307	64.96	17
+2013-03-01 09:11:58.70307	73.52	18
+2013-03-01 09:11:58.70307	78.58	19
+2013-03-01 09:11:58.70307	81.41	20
+2013-03-01 09:11:58.70307	84.71	21
+2013-03-01 09:11:58.70307	87.43	22
+2013-03-01 09:11:58.70307	91.36	23
+2013-03-01 09:11:58.70307	92.96	24
+2013-03-01 09:11:58.70307	95.04	25
+2013-03-01 09:11:58.703071	0.83	0
+2013-03-01 09:11:58.703071	1.99	1
+2013-03-01 09:11:58.703071	3.73	2
+2013-03-01 09:11:58.703071	8.86	3
+2013-03-01 09:11:58.703071	10.62	4
+2013-03-01 09:11:58.703071	11.32	5
+2013-03-01 09:11:58.703071	12.83	6
+2013-03-01 09:11:58.703071	14.7	7
+2013-03-01 09:11:58.703071	14.96	8
+2013-03-01 09:11:58.703071	17.58	9
+2013-03-01 09:11:58.703071	19.1	10
+2013-03-01 09:11:58.703071	21.01	11
+2013-03-01 09:11:58.703071	26.95	12
+2013-03-01 09:11:58.703071	27.23	13
+2013-03-01 09:11:58.703071	29.07	14
+2013-03-01 09:11:58.703071	29.71	15
+2013-03-01 09:11:58.703071	31.84	16
+2013-03-01 09:11:58.703071	31.94	17
+2013-03-01 09:11:58.703071	35.32	18
+2013-03-01 09:11:58.703071	37.32	19
+2013-03-01 09:11:58.703071	38.5	20
+2013-03-01 09:11:58.703071	42.08	21
+2013-03-01 09:11:58.703071	44.3	22
+2013-03-01 09:11:58.703071	44.66	23
+2013-03-01 09:11:58.703071	46.84	24
+2013-03-01 09:11:58.703071	48.89	25
+2013-03-01 09:11:58.703071	49.64	26
+2013-03-01 09:11:58.703071	50.28	27
+2013-03-01 09:11:58.703071	52.09	28
+2013-03-01 09:11:58.703071	53.26	29
+2013-03-01 09:11:58.703071	54.09	30
+2013-03-01 09:11:58.703071	56.45	31
+2013-03-01 09:11:58.703071	56.76	32
+2013-03-01 09:11:58.703071	61.41	33
+2013-03-01 09:11:58.703071	61.88	34
+2013-03-01 09:11:58.703071	63.03	35
+2013-03-01 09:11:58.703071	64.55	36
+2013-03-01 09:11:58.703071	68.62	37
+2013-03-01 09:11:58.703071	76.13	38
+2013-03-01 09:11:58.703071	79.05	39
+2013-03-01 09:11:58.703071	80.43	40
+2013-03-01 09:11:58.703071	81.41	41
+2013-03-01 09:11:58.703071	82.85	42
+2013-03-01 09:11:58.703071	83.98	43
+2013-03-01 09:11:58.703071	84.21	44
+2013-03-01 09:11:58.703071	85.55	45
+2013-03-01 09:11:58.703071	87.93	46
+2013-03-01 09:11:58.703071	88.93	47
+2013-03-01 09:11:58.703071	94.27	48
+2013-03-01 09:11:58.703071	99.45	49
+2013-03-01 09:11:58.703072	0.36	0
+2013-03-01 09:11:58.703072	0.48	1
+2013-03-01 09:11:58.703072	0.79	2
+2013-03-01 09:11:58.703072	1.27	3
+2013-03-01 09:11:58.703072	4.48	4
+2013-03-01 09:11:58.703072	9.0	5
+2013-03-01 09:11:58.703072	23.27	6
+2013-03-01 09:11:58.703072	25.13	7
+2013-03-01 09:11:58.703072	25.34	8
+2013-03-01 09:11:58.703072	25.91	9
+2013-03-01 09:11:58.703072	29.01	10
+2013-03-01 09:11:58.703072	30.47	11
+2013-03-01 09:11:58.703072	37.95	12
+2013-03-01 09:11:58.703072	39.3	13
+2013-03-01 09:11:58.703072	45.91	14
+2013-03-01 09:11:58.703072	52.44	15
+2013-03-01 09:11:58.703072	54.1	16
+2013-03-01 09:11:58.703072	56.7	17
+2013-03-01 09:11:58.703072	58.77	18
+2013-03-01 09:11:58.703072	62.09	19
+2013-03-01 09:11:58.703072	68.2	20
+2013-03-01 09:11:58.703072	71.68	21
+2013-03-01 09:11:58.703072	79.46	22
+2013-03-01 09:11:58.703072	80.02	23
+PREHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between 1 following and 2 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between 1 following and 2 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	3.17	2
+2013-03-01 09:11:58.70307	10.89	2
+2013-03-01 09:11:58.70307	14.54	2
+2013-03-01 09:11:58.70307	14.78	2
+2013-03-01 09:11:58.70307	17.85	2
+2013-03-01 09:11:58.70307	20.61	2
+2013-03-01 09:11:58.70307	28.69	2
+2013-03-01 09:11:58.70307	29.22	2
+2013-03-01 09:11:58.70307	31.17	2
+2013-03-01 09:11:58.70307	38.35	2
+2013-03-01 09:11:58.70307	38.61	2
+2013-03-01 09:11:58.70307	39.48	2
+2013-03-01 09:11:58.70307	40.54	2
+2013-03-01 09:11:58.70307	41.6	2
+2013-03-01 09:11:58.70307	46.08	2
+2013-03-01 09:11:58.70307	54.36	2
+2013-03-01 09:11:58.70307	56.94	2
+2013-03-01 09:11:58.70307	64.96	2
+2013-03-01 09:11:58.70307	73.52	2
+2013-03-01 09:11:58.70307	78.58	2
+2013-03-01 09:11:58.70307	81.41	2
+2013-03-01 09:11:58.70307	84.71	2
+2013-03-01 09:11:58.70307	87.43	2
+2013-03-01 09:11:58.70307	91.36	2
+2013-03-01 09:11:58.70307	92.96	1
+2013-03-01 09:11:58.70307	95.04	0
+2013-03-01 09:11:58.703071	0.83	2
+2013-03-01 09:11:58.703071	1.99	2
+2013-03-01 09:11:58.703071	3.73	2
+2013-03-01 09:11:58.703071	8.86	2
+2013-03-01 09:11:58.703071	10.62	2
+2013-03-01 09:11:58.703071	11.32	2
+2013-03-01 09:11:58.703071	12.83	2
+2013-03-01 09:11:58.703071	14.7	2
+2013-03-01 09:11:58.703071	14.96	2
+2013-03-01 09:11:58.703071	17.58	2
+2013-03-01 09:11:58.703071	19.1	2
+2013-03-01 09:11:58.703071	21.01	2
+2013-03-01 09:11:58.703071	26.95	2
+2013-03-01 09:11:58.703071	27.23	2
+2013-03-01 09:11:58.703071	29.07	2
+2013-03-01 09:11:58.703071	29.71	2
+2013-03-01 09:11:58.703071	31.84	2
+2013-03-01 09:11:58.703071	31.94	2
+2013-03-01 09:11:58.703071	35.32	2
+2013-03-01 09:11:58.703071	37.32	2
+2013-03-01 09:11:58.703071	38.5	2
+2013-03-01 09:11:58.703071	42.08	2
+2013-03-01 09:11:58.703071	44.3	2
+2013-03-01 09:11:58.703071	44.66	2
+2013-03-01 09:11:58.703071	46.84	2
+2013-03-01 09:11:58.703071	48.89	2
+2013-03-01 09:11:58.703071	49.64	2
+2013-03-01 09:11:58.703071	50.28	2
+2013-03-01 09:11:58.703071	52.09	2
+2013-03-01 09:11:58.703071	53.26	2
+2013-03-01 09:11:58.703071	54.09	2
+2013-03-01 09:11:58.703071	56.45	2
+2013-03-01 09:11:58.703071	56.76	2
+2013-03-01 09:11:58.703071	61.41	2
+2013-03-01 09:11:58.703071	61.88	2
+2013-03-01 09:11:58.703071	63.03	2
+2013-03-01 09:11:58.703071	64.55	2
+2013-03-01 09:11:58.703071	68.62	2
+2013-03-01 09:11:58.703071	76.13	2
+2013-03-01 09:11:58.703071	79.05	2
+2013-03-01 09:11:58.703071	80.43	2
+2013-03-01 09:11:58.703071	81.41	2
+2013-03-01 09:11:58.703071	82.85	2
+2013-03-01 09:11:58.703071	83.98	2
+2013-03-01 09:11:58.703071	84.21	2
+2013-03-01 09:11:58.703071	85.55	2
+2013-03-01 09:11:58.703071	87.93	2
+2013-03-01 09:11:58.703071	88.93	2
+2013-03-01 09:11:58.703071	94.27	1
+2013-03-01 09:11:58.703071	99.45	0
+2013-03-01 09:11:58.703072	0.36	2
+2013-03-01 09:11:58.703072	0.48	2
+2013-03-01 09:11:58.703072	0.79	2
+2013-03-01 09:11:58.703072	1.27	2
+2013-03-01 09:11:58.703072	4.48	2
+2013-03-01 09:11:58.703072	9.0	2
+2013-03-01 09:11:58.703072	23.27	2
+2013-03-01 09:11:58.703072	25.13	2
+2013-03-01 09:11:58.703072	25.34	2
+2013-03-01 09:11:58.703072	25.91	2
+2013-03-01 09:11:58.703072	29.01	2
+2013-03-01 09:11:58.703072	30.47	2
+2013-03-01 09:11:58.703072	37.95	2
+2013-03-01 09:11:58.703072	39.3	2
+2013-03-01 09:11:58.703072	45.91	2
+2013-03-01 09:11:58.703072	52.44	2
+2013-03-01 09:11:58.703072	54.1	2
+2013-03-01 09:11:58.703072	56.7	2
+2013-03-01 09:11:58.703072	58.77	2
+2013-03-01 09:11:58.703072	62.09	2
+2013-03-01 09:11:58.703072	68.2	2
+2013-03-01 09:11:58.703072	71.68	2
+2013-03-01 09:11:58.703072	79.46	2
+2013-03-01 09:11:58.703072	80.02	2
+PREHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, count(f) over (partition by ts order by f rows between unbounded preceding and 1 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	3.17	2
+2013-03-01 09:11:58.70307	10.89	3
+2013-03-01 09:11:58.70307	14.54	4
+2013-03-01 09:11:58.70307	14.78	5
+2013-03-01 09:11:58.70307	17.85	6
+2013-03-01 09:11:58.70307	20.61	7
+2013-03-01 09:11:58.70307	28.69	8
+2013-03-01 09:11:58.70307	29.22	9
+2013-03-01 09:11:58.70307	31.17	10
+2013-03-01 09:11:58.70307	38.35	11
+2013-03-01 09:11:58.70307	38.61	12
+2013-03-01 09:11:58.70307	39.48	13
+2013-03-01 09:11:58.70307	40.54	14
+2013-03-01 09:11:58.70307	41.6	15
+2013-03-01 09:11:58.70307	46.08	16
+2013-03-01 09:11:58.70307	54.36	17
+2013-03-01 09:11:58.70307	56.94	18
+2013-03-01 09:11:58.70307	64.96	19
+2013-03-01 09:11:58.70307	73.52	20
+2013-03-01 09:11:58.70307	78.58	21
+2013-03-01 09:11:58.70307	81.41	22
+2013-03-01 09:11:58.70307	84.71	23
+2013-03-01 09:11:58.70307	87.43	24
+2013-03-01 09:11:58.70307	91.36	25
+2013-03-01 09:11:58.70307	92.96	26
+2013-03-01 09:11:58.70307	95.04	26
+2013-03-01 09:11:58.703071	0.83	2
+2013-03-01 09:11:58.703071	1.99	3
+2013-03-01 09:11:58.703071	3.73	4
+2013-03-01 09:11:58.703071	8.86	5
+2013-03-01 09:11:58.703071	10.62	6
+2013-03-01 09:11:58.703071	11.32	7
+2013-03-01 09:11:58.703071	12.83	8
+2013-03-01 09:11:58.703071	14.7	9
+2013-03-01 09:11:58.703071	14.96	10
+2013-03-01 09:11:58.703071	17.58	11
+2013-03-01 09:11:58.703071	19.1	12
+2013-03-01 09:11:58.703071	21.01	13
+2013-03-01 09:11:58.703071	26.95	14
+2013-03-01 09:11:58.703071	27.23	15
+2013-03-01 09:11:58.703071	29.07	16
+2013-03-01 09:11:58.703071	29.71	17
+2013-03-01 09:11:58.703071	31.84	18
+2013-03-01 09:11:58.703071	31.94	19
+2013-03-01 09:11:58.703071	35.32	20
+2013-03-01 09:11:58.703071	37.32	21
+2013-03-01 09:11:58.703071	38.5	22
+2013-03-01 09:11:58.703071	42.08	23
+2013-03-01 09:11:58.703071	44.3	24
+2013-03-01 09:11:58.703071	44.66	25
+2013-03-01 09:11:58.703071	46.84	26
+2013-03-01 09:11:58.703071	48.89	27
+2013-03-01 09:11:58.703071	49.64	28
+2013-03-01 09:11:58.703071	50.28	29
+2013-03-01 09:11:58.703071	52.09	30
+2013-03-01 09:11:58.703071	53.26	31
+2013-03-01 09:11:58.703071	54.09	32
+2013-03-01 09:11:58.703071	56.45	33
+2013-03-01 09:11:58.703071	56.76	34
+2013-03-01 09:11:58.703071	61.41	35
+2013-03-01 09:11:58.703071	61.88	36
+2013-03-01 09:11:58.703071	63.03	37
+2013-03-01 09:11:58.703071	64.55	38
+2013-03-01 09:11:58.703071	68.62	39
+2013-03-01 09:11:58.703071	76.13	40
+2013-03-01 09:11:58.703071	79.05	41
+2013-03-01 09:11:58.703071	80.43	42
+2013-03-01 09:11:58.703071	81.41	43
+2013-03-01 09:11:58.703071	82.85	44
+2013-03-01 09:11:58.703071	83.98	45
+2013-03-01 09:11:58.703071	84.21	46
+2013-03-01 09:11:58.703071	85.55	47
+2013-03-01 09:11:58.703071	87.93	48
+2013-03-01 09:11:58.703071	88.93	49
+2013-03-01 09:11:58.703071	94.27	50
+2013-03-01 09:11:58.703071	99.45	50
+2013-03-01 09:11:58.703072	0.36	2
+2013-03-01 09:11:58.703072	0.48	3
+2013-03-01 09:11:58.703072	0.79	4
+2013-03-01 09:11:58.703072	1.27	5
+2013-03-01 09:11:58.703072	4.48	6
+2013-03-01 09:11:58.703072	9.0	7
+2013-03-01 09:11:58.703072	23.27	8
+2013-03-01 09:11:58.703072	25.13	9
+2013-03-01 09:11:58.703072	25.34	10
+2013-03-01 09:11:58.703072	25.91	11
+2013-03-01 09:11:58.703072	29.01	12
+2013-03-01 09:11:58.703072	30.47	13
+2013-03-01 09:11:58.703072	37.95	14
+2013-03-01 09:11:58.703072	39.3	15
+2013-03-01 09:11:58.703072	45.91	16
+2013-03-01 09:11:58.703072	52.44	17
+2013-03-01 09:11:58.703072	54.1	18
+2013-03-01 09:11:58.703072	56.7	19
+2013-03-01 09:11:58.703072	58.77	20
+2013-03-01 09:11:58.703072	62.09	21
+2013-03-01 09:11:58.703072	68.2	22
+2013-03-01 09:11:58.703072	71.68	23
+2013-03-01 09:11:58.703072	79.46	24
+2013-03-01 09:11:58.703072	80.02	25
+PREHOOK: query: -- max
+select ts, f, max(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: -- max
+select ts, f, max(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	NULL
+2013-03-01 09:11:58.70307	29.22	14.54
+2013-03-01 09:11:58.70307	39.48	29.22
+2013-03-01 09:11:58.70307	17.85	39.48
+2013-03-01 09:11:58.70307	31.17	39.48
+2013-03-01 09:11:58.70307	56.94	31.17
+2013-03-01 09:11:58.70307	78.58	56.94
+2013-03-01 09:11:58.70307	38.61	78.58
+2013-03-01 09:11:58.70307	14.78	78.58
+2013-03-01 09:11:58.70307	91.36	38.61
+2013-03-01 09:11:58.70307	28.69	91.36
+2013-03-01 09:11:58.70307	73.52	91.36
+2013-03-01 09:11:58.70307	92.96	73.52
+2013-03-01 09:11:58.70307	95.04	92.96
+2013-03-01 09:11:58.70307	41.6	95.04
+2013-03-01 09:11:58.70307	64.96	95.04
+2013-03-01 09:11:58.70307	84.71	64.96
+2013-03-01 09:11:58.70307	81.41	84.71
+2013-03-01 09:11:58.70307	10.89	84.71
+2013-03-01 09:11:58.70307	20.61	81.41
+2013-03-01 09:11:58.70307	54.36	20.61
+2013-03-01 09:11:58.70307	87.43	54.36
+2013-03-01 09:11:58.70307	46.08	87.43
+2013-03-01 09:11:58.70307	40.54	87.43
+2013-03-01 09:11:58.70307	38.35	46.08
+2013-03-01 09:11:58.70307	3.17	40.54
+2013-03-01 09:11:58.703071	0.83	NULL
+2013-03-01 09:11:58.703071	56.76	0.83
+2013-03-01 09:11:58.703071	99.45	56.76
+2013-03-01 09:11:58.703071	79.05	99.45
+2013-03-01 09:11:58.703071	14.96	99.45
+2013-03-01 09:11:58.703071	85.55	79.05
+2013-03-01 09:11:58.703071	14.7	85.55
+2013-03-01 09:11:58.703071	81.41	85.55
+2013-03-01 09:11:58.703071	87.93	81.41
+2013-03-01 09:11:58.703071	26.95	87.93
+2013-03-01 09:11:58.703071	37.32	87.93
+2013-03-01 09:11:58.703071	68.62	37.32
+2013-03-01 09:11:58.703071	80.43	68.62
+2013-03-01 09:11:58.703071	54.09	80.43
+2013-03-01 09:11:58.703071	42.08	80.43
+2013-03-01 09:11:58.703071	64.55	54.09
+2013-03-01 09:11:58.703071	48.89	64.55
+2013-03-01 09:11:58.703071	56.45	64.55
+2013-03-01 09:11:58.703071	1.99	56.45
+2013-03-01 09:11:58.703071	94.27	56.45
+2013-03-01 09:11:58.703071	35.32	94.27
+2013-03-01 09:11:58.703071	10.62	94.27
+2013-03-01 09:11:58.703071	52.09	35.32
+2013-03-01 09:11:58.703071	63.03	52.09
+2013-03-01 09:11:58.703071	76.13	63.03
+2013-03-01 09:11:58.703071	44.3	76.13
+2013-03-01 09:11:58.703071	44.66	76.13
+2013-03-01 09:11:58.703071	17.58	44.66
+2013-03-01 09:11:58.703071	11.32	44.66
+2013-03-01 09:11:58.703071	38.5	17.58
+2013-03-01 09:11:58.703071	82.85	38.5
+2013-03-01 09:11:58.703071	49.64	82.85
+2013-03-01 09:11:58.703071	27.23	82.85
+2013-03-01 09:11:58.703071	83.98	49.64
+2013-03-01 09:11:58.703071	31.84	83.98
+2013-03-01 09:11:58.703071	50.28	83.98
+2013-03-01 09:11:58.703071	3.73	50.28
+2013-03-01 09:11:58.703071	53.26	50.28
+2013-03-01 09:11:58.703071	29.71	53.26
+2013-03-01 09:11:58.703071	8.86	53.26
+2013-03-01 09:11:58.703071	21.01	29.71
+2013-03-01 09:11:58.703071	84.21	21.01
+2013-03-01 09:11:58.703071	19.1	84.21
+2013-03-01 09:11:58.703071	31.94	84.21
+2013-03-01 09:11:58.703071	88.93	31.94
+2013-03-01 09:11:58.703071	12.83	88.93
+2013-03-01 09:11:58.703071	29.07	88.93
+2013-03-01 09:11:58.703071	61.88	29.07
+2013-03-01 09:11:58.703071	61.41	61.88
+2013-03-01 09:11:58.703071	46.84	61.88
+2013-03-01 09:11:58.703072	95.01	NULL
+2013-03-01 09:11:58.703072	62.09	95.01
+2013-03-01 09:11:58.703072	29.01	95.01
+2013-03-01 09:11:58.703072	79.46	62.09
+2013-03-01 09:11:58.703072	4.48	79.46
+2013-03-01 09:11:58.703072	99.26	79.46
+2013-03-01 09:11:58.703072	58.77	99.26
+2013-03-01 09:11:58.703072	86.98	99.26
+2013-03-01 09:11:58.703072	30.47	86.98
+2013-03-01 09:11:58.703072	68.2	86.98
+2013-03-01 09:11:58.703072	52.44	68.2
+2013-03-01 09:11:58.703072	0.79	68.2
+2013-03-01 09:11:58.703072	25.34	52.44
+2013-03-01 09:11:58.703072	81.29	25.34
+2013-03-01 09:11:58.703072	71.68	81.29
+2013-03-01 09:11:58.703072	1.27	81.29
+2013-03-01 09:11:58.703072	56.7	71.68
+2013-03-01 09:11:58.703072	39.3	56.7
+2013-03-01 09:11:58.703072	25.91	56.7
+2013-03-01 09:11:58.703072	88.08	39.3
+2013-03-01 09:11:58.703072	0.48	88.08
+2013-03-01 09:11:58.703072	88.83	88.08
+2013-03-01 09:11:58.703072	9.0	88.83
+2013-03-01 09:11:58.703072	54.1	88.83
+PREHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	NULL
+2013-03-01 09:11:58.70307	29.22	14.54
+2013-03-01 09:11:58.70307	39.48	29.22
+2013-03-01 09:11:58.70307	17.85	39.48
+2013-03-01 09:11:58.70307	31.17	39.48
+2013-03-01 09:11:58.70307	56.94	39.48
+2013-03-01 09:11:58.70307	78.58	56.94
+2013-03-01 09:11:58.70307	38.61	78.58
+2013-03-01 09:11:58.70307	14.78	78.58
+2013-03-01 09:11:58.70307	91.36	78.58
+2013-03-01 09:11:58.70307	28.69	91.36
+2013-03-01 09:11:58.70307	73.52	91.36
+2013-03-01 09:11:58.70307	92.96	91.36
+2013-03-01 09:11:58.70307	95.04	92.96
+2013-03-01 09:11:58.70307	41.6	95.04
+2013-03-01 09:11:58.70307	64.96	95.04
+2013-03-01 09:11:58.70307	84.71	95.04
+2013-03-01 09:11:58.70307	81.41	95.04
+2013-03-01 09:11:58.70307	10.89	95.04
+2013-03-01 09:11:58.70307	20.61	95.04
+2013-03-01 09:11:58.70307	54.36	95.04
+2013-03-01 09:11:58.70307	87.43	95.04
+2013-03-01 09:11:58.70307	46.08	95.04
+2013-03-01 09:11:58.70307	40.54	95.04
+2013-03-01 09:11:58.70307	38.35	95.04
+2013-03-01 09:11:58.70307	3.17	95.04
+2013-03-01 09:11:58.703071	0.83	NULL
+2013-03-01 09:11:58.703071	56.76	0.83
+2013-03-01 09:11:58.703071	99.45	56.76
+2013-03-01 09:11:58.703071	79.05	99.45
+2013-03-01 09:11:58.703071	14.96	99.45
+2013-03-01 09:11:58.703071	85.55	99.45
+2013-03-01 09:11:58.703071	14.7	99.45
+2013-03-01 09:11:58.703071	81.41	99.45
+2013-03-01 09:11:58.703071	87.93	99.45
+2013-03-01 09:11:58.703071	26.95	99.45
+2013-03-01 09:11:58.703071	37.32	99.45
+2013-03-01 09:11:58.703071	68.62	99.45
+2013-03-01 09:11:58.703071	80.43	99.45
+2013-03-01 09:11:58.703071	54.09	99.45
+2013-03-01 09:11:58.703071	42.08	99.45
+2013-03-01 09:11:58.703071	64.55	99.45
+2013-03-01 09:11:58.703071	48.89	99.45
+2013-03-01 09:11:58.703071	56.45	99.45
+2013-03-01 09:11:58.703071	1.99	99.45
+2013-03-01 09:11:58.703071	94.27	99.45
+2013-03-01 09:11:58.703071	35.32	99.45
+2013-03-01 09:11:58.703071	10.62	99.45
+2013-03-01 09:11:58.703071	52.09	99.45
+2013-03-01 09:11:58.703071	63.03	99.45
+2013-03-01 09:11:58.703071	76.13	99.45
+2013-03-01 09:11:58.703071	44.3	99.45
+2013-03-01 09:11:58.703071	44.66	99.45
+2013-03-01 09:11:58.703071	17.58	99.45
+2013-03-01 09:11:58.703071	11.32	99.45
+2013-03-01 09:11:58.703071	38.5	99.45
+2013-03-01 09:11:58.703071	82.85	99.45
+2013-03-01 09:11:58.703071	49.64	99.45
+2013-03-01 09:11:58.703071	27.23	99.45
+2013-03-01 09:11:58.703071	83.98	99.45
+2013-03-01 09:11:58.703071	31.84	99.45
+2013-03-01 09:11:58.703071	50.28	99.45
+2013-03-01 09:11:58.703071	3.73	99.45
+2013-03-01 09:11:58.703071	53.26	99.45
+2013-03-01 09:11:58.703071	29.71	99.45
+2013-03-01 09:11:58.703071	8.86	99.45
+2013-03-01 09:11:58.703071	21.01	99.45
+2013-03-01 09:11:58.703071	84.21	99.45
+2013-03-01 09:11:58.703071	19.1	99.45
+2013-03-01 09:11:58.703071	31.94	99.45
+2013-03-01 09:11:58.703071	88.93	99.45
+2013-03-01 09:11:58.703071	12.83	99.45
+2013-03-01 09:11:58.703071	29.07	99.45
+2013-03-01 09:11:58.703071	61.88	99.45
+2013-03-01 09:11:58.703071	61.41	99.45
+2013-03-01 09:11:58.703071	46.84	99.45
+2013-03-01 09:11:58.703072	95.01	NULL
+2013-03-01 09:11:58.703072	62.09	95.01
+2013-03-01 09:11:58.703072	29.01	95.01
+2013-03-01 09:11:58.703072	79.46	95.01
+2013-03-01 09:11:58.703072	4.48	95.01
+2013-03-01 09:11:58.703072	99.26	95.01
+2013-03-01 09:11:58.703072	58.77	99.26
+2013-03-01 09:11:58.703072	86.98	99.26
+2013-03-01 09:11:58.703072	30.47	99.26
+2013-03-01 09:11:58.703072	68.2	99.26
+2013-03-01 09:11:58.703072	52.44	99.26
+2013-03-01 09:11:58.703072	0.79	99.26
+2013-03-01 09:11:58.703072	25.34	99.26
+2013-03-01 09:11:58.703072	81.29	99.26
+2013-03-01 09:11:58.703072	71.68	99.26
+2013-03-01 09:11:58.703072	1.27	99.26
+2013-03-01 09:11:58.703072	56.7	99.26
+2013-03-01 09:11:58.703072	39.3	99.26
+2013-03-01 09:11:58.703072	25.91	99.26
+2013-03-01 09:11:58.703072	88.08	99.26
+2013-03-01 09:11:58.703072	0.48	99.26
+2013-03-01 09:11:58.703072	88.83	99.26
+2013-03-01 09:11:58.703072	9.0	99.26
+2013-03-01 09:11:58.703072	54.1	99.26
+PREHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	39.48
+2013-03-01 09:11:58.70307	29.22	39.48
+2013-03-01 09:11:58.70307	39.48	31.17
+2013-03-01 09:11:58.70307	17.85	56.94
+2013-03-01 09:11:58.70307	31.17	78.58
+2013-03-01 09:11:58.70307	56.94	78.58
+2013-03-01 09:11:58.70307	78.58	38.61
+2013-03-01 09:11:58.70307	38.61	91.36
+2013-03-01 09:11:58.70307	14.78	91.36
+2013-03-01 09:11:58.70307	91.36	73.52
+2013-03-01 09:11:58.70307	28.69	92.96
+2013-03-01 09:11:58.70307	73.52	95.04
+2013-03-01 09:11:58.70307	92.96	95.04
+2013-03-01 09:11:58.70307	95.04	64.96
+2013-03-01 09:11:58.70307	41.6	84.71
+2013-03-01 09:11:58.70307	64.96	84.71
+2013-03-01 09:11:58.70307	84.71	81.41
+2013-03-01 09:11:58.70307	81.41	20.61
+2013-03-01 09:11:58.70307	10.89	54.36
+2013-03-01 09:11:58.70307	20.61	87.43
+2013-03-01 09:11:58.70307	54.36	87.43
+2013-03-01 09:11:58.70307	87.43	46.08
+2013-03-01 09:11:58.70307	46.08	40.54
+2013-03-01 09:11:58.70307	40.54	38.35
+2013-03-01 09:11:58.70307	38.35	3.17
+2013-03-01 09:11:58.70307	3.17	NULL
+2013-03-01 09:11:58.703071	0.83	99.45
+2013-03-01 09:11:58.703071	56.76	99.45
+2013-03-01 09:11:58.703071	99.45	79.05
+2013-03-01 09:11:58.703071	79.05	85.55
+2013-03-01 09:11:58.703071	14.96	85.55
+2013-03-01 09:11:58.703071	85.55	81.41
+2013-03-01 09:11:58.703071	14.7	87.93
+2013-03-01 09:11:58.703071	81.41	87.93
+2013-03-01 09:11:58.703071	87.93	37.32
+2013-03-01 09:11:58.703071	26.95	68.62
+2013-03-01 09:11:58.703071	37.32	80.43
+2013-03-01 09:11:58.703071	68.62	80.43
+2013-03-01 09:11:58.703071	80.43	54.09
+2013-03-01 09:11:58.703071	54.09	64.55
+2013-03-01 09:11:58.703071	42.08	64.55
+2013-03-01 09:11:58.703071	64.55	56.45
+2013-03-01 09:11:58.703071	48.89	56.45
+2013-03-01 09:11:58.703071	56.45	94.27
+2013-03-01 09:11:58.703071	1.99	94.27
+2013-03-01 09:11:58.703071	94.27	35.32
+2013-03-01 09:11:58.703071	35.32	52.09
+2013-03-01 09:11:58.703071	10.62	63.03
+2013-03-01 09:11:58.703071	52.09	76.13
+2013-03-01 09:11:58.703071	63.03	76.13
+2013-03-01 09:11:58.703071	76.13	44.66
+2013-03-01 09:11:58.703071	44.3	44.66
+2013-03-01 09:11:58.703071	44.66	17.58
+2013-03-01 09:11:58.703071	17.58	38.5
+2013-03-01 09:11:58.703071	11.32	82.85
+2013-03-01 09:11:58.703071	38.5	82.85
+2013-03-01 09:11:58.703071	82.85	49.64
+2013-03-01 09:11:58.703071	49.64	83.98
+2013-03-01 09:11:58.703071	27.23	83.98
+2013-03-01 09:11:58.703071	83.98	50.28
+2013-03-01 09:11:58.703071	31.84	50.28
+2013-03-01 09:11:58.703071	50.28	53.26
+2013-03-01 09:11:58.703071	3.73	53.26
+2013-03-01 09:11:58.703071	53.26	29.71
+2013-03-01 09:11:58.703071	29.71	21.01
+2013-03-01 09:11:58.703071	8.86	84.21
+2013-03-01 09:11:58.703071	21.01	84.21
+2013-03-01 09:11:58.703071	84.21	31.94
+2013-03-01 09:11:58.703071	19.1	88.93
+2013-03-01 09:11:58.703071	31.94	88.93
+2013-03-01 09:11:58.703071	88.93	29.07
+2013-03-01 09:11:58.703071	12.83	61.88
+2013-03-01 09:11:58.703071	29.07	61.88
+2013-03-01 09:11:58.703071	61.88	61.41
+2013-03-01 09:11:58.703071	61.41	46.84
+2013-03-01 09:11:58.703071	46.84	NULL
+2013-03-01 09:11:58.703072	95.01	62.09
+2013-03-01 09:11:58.703072	62.09	79.46
+2013-03-01 09:11:58.703072	29.01	79.46
+2013-03-01 09:11:58.703072	79.46	99.26
+2013-03-01 09:11:58.703072	4.48	99.26
+2013-03-01 09:11:58.703072	99.26	86.98
+2013-03-01 09:11:58.703072	58.77	86.98
+2013-03-01 09:11:58.703072	86.98	68.2
+2013-03-01 09:11:58.703072	30.47	68.2
+2013-03-01 09:11:58.703072	68.2	52.44
+2013-03-01 09:11:58.703072	52.44	25.34
+2013-03-01 09:11:58.703072	0.79	81.29
+2013-03-01 09:11:58.703072	25.34	81.29
+2013-03-01 09:11:58.703072	81.29	71.68
+2013-03-01 09:11:58.703072	71.68	56.7
+2013-03-01 09:11:58.703072	1.27	56.7
+2013-03-01 09:11:58.703072	56.7	39.3
+2013-03-01 09:11:58.703072	39.3	88.08
+2013-03-01 09:11:58.703072	25.91	88.08
+2013-03-01 09:11:58.703072	88.08	88.83
+2013-03-01 09:11:58.703072	0.48	88.83
+2013-03-01 09:11:58.703072	88.83	54.1
+2013-03-01 09:11:58.703072	9.0	54.1
+2013-03-01 09:11:58.703072	54.1	45.91
+PREHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, max(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	29.22
+2013-03-01 09:11:58.70307	29.22	39.48
+2013-03-01 09:11:58.70307	39.48	39.48
+2013-03-01 09:11:58.70307	17.85	39.48
+2013-03-01 09:11:58.70307	31.17	56.94
+2013-03-01 09:11:58.70307	56.94	78.58
+2013-03-01 09:11:58.70307	78.58	78.58
+2013-03-01 09:11:58.70307	38.61	78.58
+2013-03-01 09:11:58.70307	14.78	91.36
+2013-03-01 09:11:58.70307	91.36	91.36
+2013-03-01 09:11:58.70307	28.69	91.36
+2013-03-01 09:11:58.70307	73.52	92.96
+2013-03-01 09:11:58.70307	92.96	95.04
+2013-03-01 09:11:58.70307	95.04	95.04
+2013-03-01 09:11:58.70307	41.6	95.04
+2013-03-01 09:11:58.70307	64.96	95.04
+2013-03-01 09:11:58.70307	84.71	95.04
+2013-03-01 09:11:58.70307	81.41	95.04
+2013-03-01 09:11:58.70307	10.89	95.04
+2013-03-01 09:11:58.70307	20.61	95.04
+2013-03-01 09:11:58.70307	54.36	95.04
+2013-03-01 09:11:58.70307	87.43	95.04
+2013-03-01 09:11:58.70307	46.08	95.04
+2013-03-01 09:11:58.70307	40.54	95.04
+2013-03-01 09:11:58.70307	38.35	95.04
+2013-03-01 09:11:58.70307	3.17	95.04
+2013-03-01 09:11:58.703071	0.83	56.76
+2013-03-01 09:11:58.703071	56.76	99.45
+2013-03-01 09:11:58.703071	99.45	99.45
+2013-03-01 09:11:58.703071	79.05	99.45
+2013-03-01 09:11:58.703071	14.96	99.45
+2013-03-01 09:11:58.703071	85.55	99.45
+2013-03-01 09:11:58.703071	14.7	99.45
+2013-03-01 09:11:58.703071	81.41	99.45
+2013-03-01 09:11:58.703071	87.93	99.45
+2013-03-01 09:11:58.703071	26.95	99.45
+2013-03-01 09:11:58.703071	37.32	99.45
+2013-03-01 09:11:58.703071	68.62	99.45
+2013-03-01 09:11:58.703071	80.43	99.45
+2013-03-01 09:11:58.703071	54.09	99.45
+2013-03-01 09:11:58.703071	42.08	99.45
+2013-03-01 09:11:58.703071	64.55	99.45
+2013-03-01 09:11:58.703071	48.89	99.45
+2013-03-01 09:11:58.703071	56.45	99.45
+2013-03-01 09:11:58.703071	1.99	99.45
+2013-03-01 09:11:58.703071	94.27	99.45
+2013-03-01 09:11:58.703071	35.32	99.45
+2013-03-01 09:11:58.703071	10.62	99.45
+2013-03-01 09:11:58.703071	52.09	99.45
+2013-03-01 09:11:58.703071	63.03	99.45
+2013-03-01 09:11:58.703071	76.13	99.45
+2013-03-01 09:11:58.703071	44.3	99.45
+2013-03-01 09:11:58.703071	44.66	99.45
+2013-03-01 09:11:58.703071	17.58	99.45
+2013-03-01 09:11:58.703071	11.32	99.45
+2013-03-01 09:11:58.703071	38.5	99.45
+2013-03-01 09:11:58.703071	82.85	99.45
+2013-03-01 09:11:58.703071	49.64	99.45
+2013-03-01 09:11:58.703071	27.23	99.45
+2013-03-01 09:11:58.703071	83.98	99.45
+2013-03-01 09:11:58.703071	31.84	99.45
+2013-03-01 09:11:58.703071	50.28	99.45
+2013-03-01 09:11:58.703071	3.73	99.45
+2013-03-01 09:11:58.703071	53.26	99.45
+2013-03-01 09:11:58.703071	29.71	99.45
+2013-03-01 09:11:58.703071	8.86	99.45
+2013-03-01 09:11:58.703071	21.01	99.45
+2013-03-01 09:11:58.703071	84.21	99.45
+2013-03-01 09:11:58.703071	19.1	99.45
+2013-03-01 09:11:58.703071	31.94	99.45
+2013-03-01 09:11:58.703071	88.93	99.45
+2013-03-01 09:11:58.703071	12.83	99.45
+2013-03-01 09:11:58.703071	29.07	99.45
+2013-03-01 09:11:58.703071	61.88	99.45
+2013-03-01 09:11:58.703071	61.41	99.45
+2013-03-01 09:11:58.703071	46.84	99.45
+2013-03-01 09:11:58.703072	95.01	95.01
+2013-03-01 09:11:58.703072	62.09	95.01
+2013-03-01 09:11:58.703072	29.01	95.01
+2013-03-01 09:11:58.703072	79.46	95.01
+2013-03-01 09:11:58.703072	4.48	99.26
+2013-03-01 09:11:58.703072	99.26	99.26
+2013-03-01 09:11:58.703072	58.77	99.26
+2013-03-01 09:11:58.703072	86.98	99.26
+2013-03-01 09:11:58.703072	30.47	99.26
+2013-03-01 09:11:58.703072	68.2	99.26
+2013-03-01 09:11:58.703072	52.44	99.26
+2013-03-01 09:11:58.703072	0.79	99.26
+2013-03-01 09:11:58.703072	25.34	99.26
+2013-03-01 09:11:58.703072	81.29	99.26
+2013-03-01 09:11:58.703072	71.68	99.26
+2013-03-01 09:11:58.703072	1.27	99.26
+2013-03-01 09:11:58.703072	56.7	99.26
+2013-03-01 09:11:58.703072	39.3	99.26
+2013-03-01 09:11:58.703072	25.91	99.26
+2013-03-01 09:11:58.703072	88.08	99.26
+2013-03-01 09:11:58.703072	0.48	99.26
+2013-03-01 09:11:58.703072	88.83	99.26
+2013-03-01 09:11:58.703072	9.0	99.26
+2013-03-01 09:11:58.703072	54.1	99.26
+PREHOOK: query: -- min
+select ts, f, min(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: -- min
+select ts, f, min(f) over (partition by ts order by t rows between 2 preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	NULL
+2013-03-01 09:11:58.70307	29.22	14.54
+2013-03-01 09:11:58.70307	39.48	14.54
+2013-03-01 09:11:58.70307	17.85	29.22
+2013-03-01 09:11:58.70307	31.17	17.85
+2013-03-01 09:11:58.70307	56.94	17.85
+2013-03-01 09:11:58.70307	78.58	31.17
+2013-03-01 09:11:58.70307	38.61	56.94
+2013-03-01 09:11:58.70307	14.78	38.61
+2013-03-01 09:11:58.70307	91.36	14.78
+2013-03-01 09:11:58.70307	28.69	14.78
+2013-03-01 09:11:58.70307	73.52	28.69
+2013-03-01 09:11:58.70307	92.96	28.69
+2013-03-01 09:11:58.70307	95.04	73.52
+2013-03-01 09:11:58.70307	41.6	92.96
+2013-03-01 09:11:58.70307	64.96	41.6
+2013-03-01 09:11:58.70307	84.71	41.6
+2013-03-01 09:11:58.70307	81.41	64.96
+2013-03-01 09:11:58.70307	10.89	81.41
+2013-03-01 09:11:58.70307	20.61	10.89
+2013-03-01 09:11:58.70307	54.36	10.89
+2013-03-01 09:11:58.70307	87.43	20.61
+2013-03-01 09:11:58.70307	46.08	54.36
+2013-03-01 09:11:58.70307	40.54	46.08
+2013-03-01 09:11:58.70307	38.35	40.54
+2013-03-01 09:11:58.70307	3.17	38.35
+2013-03-01 09:11:58.703071	0.83	NULL
+2013-03-01 09:11:58.703071	56.76	0.83
+2013-03-01 09:11:58.703071	99.45	0.83
+2013-03-01 09:11:58.703071	79.05	56.76
+2013-03-01 09:11:58.703071	14.96	79.05
+2013-03-01 09:11:58.703071	85.55	14.96
+2013-03-01 09:11:58.703071	14.7	14.96
+2013-03-01 09:11:58.703071	81.41	14.7
+2013-03-01 09:11:58.703071	87.93	14.7
+2013-03-01 09:11:58.703071	26.95	81.41
+2013-03-01 09:11:58.703071	37.32	26.95
+2013-03-01 09:11:58.703071	68.62	26.95
+2013-03-01 09:11:58.703071	80.43	37.32
+2013-03-01 09:11:58.703071	54.09	68.62
+2013-03-01 09:11:58.703071	42.08	54.09
+2013-03-01 09:11:58.703071	64.55	42.08
+2013-03-01 09:11:58.703071	48.89	42.08
+2013-03-01 09:11:58.703071	56.45	48.89
+2013-03-01 09:11:58.703071	1.99	48.89
+2013-03-01 09:11:58.703071	94.27	1.99
+2013-03-01 09:11:58.703071	35.32	1.99
+2013-03-01 09:11:58.703071	10.62	35.32
+2013-03-01 09:11:58.703071	52.09	10.62
+2013-03-01 09:11:58.703071	63.03	10.62
+2013-03-01 09:11:58.703071	76.13	52.09
+2013-03-01 09:11:58.703071	44.3	63.03
+2013-03-01 09:11:58.703071	44.66	44.3
+2013-03-01 09:11:58.703071	17.58	44.3
+2013-03-01 09:11:58.703071	11.32	17.58
+2013-03-01 09:11:58.703071	38.5	11.32
+2013-03-01 09:11:58.703071	82.85	11.32
+2013-03-01 09:11:58.703071	49.64	38.5
+2013-03-01 09:11:58.703071	27.23	49.64
+2013-03-01 09:11:58.703071	83.98	27.23
+2013-03-01 09:11:58.703071	31.84	27.23
+2013-03-01 09:11:58.703071	50.28	31.84
+2013-03-01 09:11:58.703071	3.73	31.84
+2013-03-01 09:11:58.703071	53.26	3.73
+2013-03-01 09:11:58.703071	29.71	3.73
+2013-03-01 09:11:58.703071	8.86	29.71
+2013-03-01 09:11:58.703071	21.01	8.86
+2013-03-01 09:11:58.703071	84.21	8.86
+2013-03-01 09:11:58.703071	19.1	21.01
+2013-03-01 09:11:58.703071	31.94	19.1
+2013-03-01 09:11:58.703071	88.93	19.1
+2013-03-01 09:11:58.703071	12.83	31.94
+2013-03-01 09:11:58.703071	29.07	12.83
+2013-03-01 09:11:58.703071	61.88	12.83
+2013-03-01 09:11:58.703071	61.41	29.07
+2013-03-01 09:11:58.703071	46.84	61.41
+2013-03-01 09:11:58.703072	95.01	NULL
+2013-03-01 09:11:58.703072	62.09	95.01
+2013-03-01 09:11:58.703072	29.01	62.09
+2013-03-01 09:11:58.703072	79.46	29.01
+2013-03-01 09:11:58.703072	4.48	29.01
+2013-03-01 09:11:58.703072	99.26	4.48
+2013-03-01 09:11:58.703072	58.77	4.48
+2013-03-01 09:11:58.703072	86.98	58.77
+2013-03-01 09:11:58.703072	30.47	58.77
+2013-03-01 09:11:58.703072	68.2	30.47
+2013-03-01 09:11:58.703072	52.44	30.47
+2013-03-01 09:11:58.703072	0.79	52.44
+2013-03-01 09:11:58.703072	25.34	0.79
+2013-03-01 09:11:58.703072	81.29	0.79
+2013-03-01 09:11:58.703072	71.68	25.34
+2013-03-01 09:11:58.703072	1.27	71.68
+2013-03-01 09:11:58.703072	56.7	1.27
+2013-03-01 09:11:58.703072	39.3	1.27
+2013-03-01 09:11:58.703072	25.91	39.3
+2013-03-01 09:11:58.703072	88.08	25.91
+2013-03-01 09:11:58.703072	0.48	25.91
+2013-03-01 09:11:58.703072	88.83	0.48
+2013-03-01 09:11:58.703072	9.0	0.48
+2013-03-01 09:11:58.703072	54.1	9.0
+PREHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 preceding) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	NULL
+2013-03-01 09:11:58.70307	29.22	14.54
+2013-03-01 09:11:58.70307	39.48	14.54
+2013-03-01 09:11:58.70307	17.85	14.54
+2013-03-01 09:11:58.70307	31.17	14.54
+2013-03-01 09:11:58.70307	56.94	14.54
+2013-03-01 09:11:58.70307	78.58	14.54
+2013-03-01 09:11:58.70307	38.61	14.54
+2013-03-01 09:11:58.70307	14.78	14.54
+2013-03-01 09:11:58.70307	91.36	14.54
+2013-03-01 09:11:58.70307	28.69	14.54
+2013-03-01 09:11:58.70307	73.52	14.54
+2013-03-01 09:11:58.70307	92.96	14.54
+2013-03-01 09:11:58.70307	95.04	14.54
+2013-03-01 09:11:58.70307	41.6	14.54
+2013-03-01 09:11:58.70307	64.96	14.54
+2013-03-01 09:11:58.70307	84.71	14.54
+2013-03-01 09:11:58.70307	81.41	14.54
+2013-03-01 09:11:58.70307	10.89	14.54
+2013-03-01 09:11:58.70307	20.61	10.89
+2013-03-01 09:11:58.70307	54.36	10.89
+2013-03-01 09:11:58.70307	87.43	10.89
+2013-03-01 09:11:58.70307	46.08	10.89
+2013-03-01 09:11:58.70307	40.54	10.89
+2013-03-01 09:11:58.70307	38.35	10.89
+2013-03-01 09:11:58.70307	3.17	10.89
+2013-03-01 09:11:58.703071	0.83	NULL
+2013-03-01 09:11:58.703071	56.76	0.83
+2013-03-01 09:11:58.703071	99.45	0.83
+2013-03-01 09:11:58.703071	79.05	0.83
+2013-03-01 09:11:58.703071	14.96	0.83
+2013-03-01 09:11:58.703071	85.55	0.83
+2013-03-01 09:11:58.703071	14.7	0.83
+2013-03-01 09:11:58.703071	81.41	0.83
+2013-03-01 09:11:58.703071	87.93	0.83
+2013-03-01 09:11:58.703071	26.95	0.83
+2013-03-01 09:11:58.703071	37.32	0.83
+2013-03-01 09:11:58.703071	68.62	0.83
+2013-03-01 09:11:58.703071	80.43	0.83
+2013-03-01 09:11:58.703071	54.09	0.83
+2013-03-01 09:11:58.703071	42.08	0.83
+2013-03-01 09:11:58.703071	64.55	0.83
+2013-03-01 09:11:58.703071	48.89	0.83
+2013-03-01 09:11:58.703071	56.45	0.83
+2013-03-01 09:11:58.703071	1.99	0.83
+2013-03-01 09:11:58.703071	94.27	0.83
+2013-03-01 09:11:58.703071	35.32	0.83
+2013-03-01 09:11:58.703071	10.62	0.83
+2013-03-01 09:11:58.703071	52.09	0.83
+2013-03-01 09:11:58.703071	63.03	0.83
+2013-03-01 09:11:58.703071	76.13	0.83
+2013-03-01 09:11:58.703071	44.3	0.83
+2013-03-01 09:11:58.703071	44.66	0.83
+2013-03-01 09:11:58.703071	17.58	0.83
+2013-03-01 09:11:58.703071	11.32	0.83
+2013-03-01 09:11:58.703071	38.5	0.83
+2013-03-01 09:11:58.703071	82.85	0.83
+2013-03-01 09:11:58.703071	49.64	0.83
+2013-03-01 09:11:58.703071	27.23	0.83
+2013-03-01 09:11:58.703071	83.98	0.83
+2013-03-01 09:11:58.703071	31.84	0.83
+2013-03-01 09:11:58.703071	50.28	0.83
+2013-03-01 09:11:58.703071	3.73	0.83
+2013-03-01 09:11:58.703071	53.26	0.83
+2013-03-01 09:11:58.703071	29.71	0.83
+2013-03-01 09:11:58.703071	8.86	0.83
+2013-03-01 09:11:58.703071	21.01	0.83
+2013-03-01 09:11:58.703071	84.21	0.83
+2013-03-01 09:11:58.703071	19.1	0.83
+2013-03-01 09:11:58.703071	31.94	0.83
+2013-03-01 09:11:58.703071	88.93	0.83
+2013-03-01 09:11:58.703071	12.83	0.83
+2013-03-01 09:11:58.703071	29.07	0.83
+2013-03-01 09:11:58.703071	61.88	0.83
+2013-03-01 09:11:58.703071	61.41	0.83
+2013-03-01 09:11:58.703071	46.84	0.83
+2013-03-01 09:11:58.703072	95.01	NULL
+2013-03-01 09:11:58.703072	62.09	95.01
+2013-03-01 09:11:58.703072	29.01	62.09
+2013-03-01 09:11:58.703072	79.46	29.01
+2013-03-01 09:11:58.703072	4.48	29.01
+2013-03-01 09:11:58.703072	99.26	4.48
+2013-03-01 09:11:58.703072	58.77	4.48
+2013-03-01 09:11:58.703072	86.98	4.48
+2013-03-01 09:11:58.703072	30.47	4.48
+2013-03-01 09:11:58.703072	68.2	4.48
+2013-03-01 09:11:58.703072	52.44	4.48
+2013-03-01 09:11:58.703072	0.79	4.48
+2013-03-01 09:11:58.703072	25.34	0.79
+2013-03-01 09:11:58.703072	81.29	0.79
+2013-03-01 09:11:58.703072	71.68	0.79
+2013-03-01 09:11:58.703072	1.27	0.79
+2013-03-01 09:11:58.703072	56.7	0.79
+2013-03-01 09:11:58.703072	39.3	0.79
+2013-03-01 09:11:58.703072	25.91	0.79
+2013-03-01 09:11:58.703072	88.08	0.79
+2013-03-01 09:11:58.703072	0.48	0.79
+2013-03-01 09:11:58.703072	88.83	0.48
+2013-03-01 09:11:58.703072	9.0	0.48
+2013-03-01 09:11:58.703072	54.1	0.48
+PREHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between 1 following and 2 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	29.22
+2013-03-01 09:11:58.70307	29.22	17.85
+2013-03-01 09:11:58.70307	39.48	17.85
+2013-03-01 09:11:58.70307	17.85	31.17
+2013-03-01 09:11:58.70307	31.17	56.94
+2013-03-01 09:11:58.70307	56.94	38.61
+2013-03-01 09:11:58.70307	78.58	14.78
+2013-03-01 09:11:58.70307	38.61	14.78
+2013-03-01 09:11:58.70307	14.78	28.69
+2013-03-01 09:11:58.70307	91.36	28.69
+2013-03-01 09:11:58.70307	28.69	73.52
+2013-03-01 09:11:58.70307	73.52	92.96
+2013-03-01 09:11:58.70307	92.96	41.6
+2013-03-01 09:11:58.70307	95.04	41.6
+2013-03-01 09:11:58.70307	41.6	64.96
+2013-03-01 09:11:58.70307	64.96	81.41
+2013-03-01 09:11:58.70307	84.71	10.89
+2013-03-01 09:11:58.70307	81.41	10.89
+2013-03-01 09:11:58.70307	10.89	20.61
+2013-03-01 09:11:58.70307	20.61	54.36
+2013-03-01 09:11:58.70307	54.36	46.08
+2013-03-01 09:11:58.70307	87.43	40.54
+2013-03-01 09:11:58.70307	46.08	38.35
+2013-03-01 09:11:58.70307	40.54	3.17
+2013-03-01 09:11:58.70307	38.35	3.17
+2013-03-01 09:11:58.70307	3.17	NULL
+2013-03-01 09:11:58.703071	0.83	56.76
+2013-03-01 09:11:58.703071	56.76	79.05
+2013-03-01 09:11:58.703071	99.45	14.96
+2013-03-01 09:11:58.703071	79.05	14.96
+2013-03-01 09:11:58.703071	14.96	14.7
+2013-03-01 09:11:58.703071	85.55	14.7
+2013-03-01 09:11:58.703071	14.7	81.41
+2013-03-01 09:11:58.703071	81.41	26.95
+2013-03-01 09:11:58.703071	87.93	26.95
+2013-03-01 09:11:58.703071	26.95	37.32
+2013-03-01 09:11:58.703071	37.32	68.62
+2013-03-01 09:11:58.703071	68.62	54.09
+2013-03-01 09:11:58.703071	80.43	42.08
+2013-03-01 09:11:58.703071	54.09	42.08
+2013-03-01 09:11:58.703071	42.08	48.89
+2013-03-01 09:11:58.703071	64.55	48.89
+2013-03-01 09:11:58.703071	48.89	1.99
+2013-03-01 09:11:58.703071	56.45	1.99
+2013-03-01 09:11:58.703071	1.99	35.32
+2013-03-01 09:11:58.703071	94.27	10.62
+2013-03-01 09:11:58.703071	35.32	10.62
+2013-03-01 09:11:58.703071	10.62	52.09
+2013-03-01 09:11:58.703071	52.09	63.03
+2013-03-01 09:11:58.703071	63.03	44.3
+2013-03-01 09:11:58.703071	76.13	44.3
+2013-03-01 09:11:58.703071	44.3	17.58
+2013-03-01 09:11:58.703071	44.66	11.32
+2013-03-01 09:11:58.703071	17.58	11.32
+2013-03-01 09:11:58.703071	11.32	38.5
+2013-03-01 09:11:58.703071	38.5	49.64
+2013-03-01 09:11:58.703071	82.85	27.23
+2013-03-01 09:11:58.703071	49.64	27.23
+2013-03-01 09:11:58.703071	27.23	31.84
+2013-03-01 09:11:58.703071	83.98	31.84
+2013-03-01 09:11:58.703071	31.84	3.73
+2013-03-01 09:11:58.703071	50.28	3.73
+2013-03-01 09:11:58.703071	3.73	29.71
+2013-03-01 09:11:58.703071	53.26	8.86
+2013-03-01 09:11:58.703071	29.71	8.86
+2013-03-01 09:11:58.703071	8.86	21.01
+2013-03-01 09:11:58.703071	21.01	19.1
+2013-03-01 09:11:58.703071	84.21	19.1
+2013-03-01 09:11:58.703071	19.1	31.94
+2013-03-01 09:11:58.703071	31.94	12.83
+2013-03-01 09:11:58.703071	88.93	12.83
+2013-03-01 09:11:58.703071	12.83	29.07
+2013-03-01 09:11:58.703071	29.07	61.41
+2013-03-01 09:11:58.703071	61.88	46.84
+2013-03-01 09:11:58.703071	61.41	46.84
+2013-03-01 09:11:58.703071	46.84	NULL
+2013-03-01 09:11:58.703072	95.01	29.01
+2013-03-01 09:11:58.703072	62.09	29.01
+2013-03-01 09:11:58.703072	29.01	4.48
+2013-03-01 09:11:58.703072	79.46	4.48
+2013-03-01 09:11:58.703072	4.48	58.77
+2013-03-01 09:11:58.703072	99.26	58.77
+2013-03-01 09:11:58.703072	58.77	30.47
+2013-03-01 09:11:58.703072	86.98	30.47
+2013-03-01 09:11:58.703072	30.47	52.44
+2013-03-01 09:11:58.703072	68.2	0.79
+2013-03-01 09:11:58.703072	52.44	0.79
+2013-03-01 09:11:58.703072	0.79	25.34
+2013-03-01 09:11:58.703072	25.34	71.68
+2013-03-01 09:11:58.703072	81.29	1.27
+2013-03-01 09:11:58.703072	71.68	1.27
+2013-03-01 09:11:58.703072	1.27	39.3
+2013-03-01 09:11:58.703072	56.7	25.91
+2013-03-01 09:11:58.703072	39.3	25.91
+2013-03-01 09:11:58.703072	25.91	0.48
+2013-03-01 09:11:58.703072	88.08	0.48
+2013-03-01 09:11:58.703072	0.48	9.0
+2013-03-01 09:11:58.703072	88.83	9.0
+2013-03-01 09:11:58.703072	9.0	45.91
+2013-03-01 09:11:58.703072	54.1	0.36
+PREHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100
+PREHOOK: type: QUERY
+PREHOOK: Input: default@over10k
+#### A masked pattern was here ####
+POSTHOOK: query: select ts, f, min(f) over (partition by ts order by t rows between unbounded preceding and 1 following) from over10k limit 100
+POSTHOOK: type: QUERY
+POSTHOOK: Input: default@over10k
+#### A masked pattern was here ####
+2013-03-01 09:11:58.70307	14.54	14.54
+2013-03-01 09:11:58.70307	29.22	14.54
+2013-03-01 09:11:58.70307	39.48	14.54
+2013-03-01 09:11:58.70307	17.85	14.54
+2013-03-01 09:11:58.70307	31.17	14.54
+2013-03-01 09:11:58.70307	56.94	14.54
+2013-03-01 09:11:58.70307	78.58	14.54
+2013-03-01 09:11:58.70307	38.61	14.54
+2013-03-01 09:11:58.70307	14.78	14.54
+2013-03-01 09:11:58.70307	91.36	14.54
+2013-03-01 09:11:58.70307	28.69	14.54
+2013-03-01 09:11:58.70307	73.52	14.54
+2013-03-01 09:11:58.70307	92.96	14.54
+2013-03-01 09:11:58.70307	95.04	14.54
+2013-03-01 09:11:58.70307	41.6	14.54
+2013-03-01 09:11:58.70307	64.96	14.54
+2013-03-01 09:11:58.70307	84.71	14.54
+2013-03-01 09:11:58.70307	81.41	10.89
+2013-03-01 09:11:58.70307	10.89	10.89
+2013-03-01 09:11:58.70307	20.61	10.89
+2013-03-01 09:11:58.70307	54.36	10.89
+2013-03-01 09:11:58.70307	87.43	10.89
+2013-03-01 09:11:58.70307	46.08	10.89
+2013-03-01 09:11:58.70307	40.54	10.89
+2013-03-01 09:11:58.70307	38.35	3.17
+2013-03-01 09:11:58.70307	3.17	3.17
+2013-03-01 09:11:58.703071	0.83	0.83
+2013-03-01 09:11:58.703071	56.76	0.83
+2013-03-01 09:11:58.703071	99.45	0.83
+2013-03-01 09:11:58.703071	79.05	0.83
+2013-03-01 09:11:58.703071	14.96	0.83
+2013-03-01 09:11:58.703071	85.55	0.83
+2013-03-01 09:11:58.703071	14.7	0.83
+2013-03-01 09:11:58.703071	81.41	0.83
+2013-03-01 09:11:58.703071	87.93	0.83
+2013-03-01 09:11:58.703071	26.95	0.83
+2013-03-01 09:11:58.703071	37.32	0.83
+2013-03-01 09:11:58.703071	68.62	0.83
+2013-03-01 09:11:58.703071	80.43	0.83
+2013-03-01 09:11:58.703071	54.09	0.83
+2013-03-01 09:11:58.703071	42.08	0.83
+2013-03-01 09:11:58.703071	64.55	0.83
+2013-03-01 09:11:58.703071	48.89	0.83
+2013-03-01 09:11:58.703071	56.45	0.83
+2013-03-01 09:11:58.703071	1.99	0.83
+2013-03-01 09:11:58.703071	94.27	0.83
+2013-03-01 09:11:58.703071	35.32	0.83
+2013-03-01 09:11:58.703071	10.62	0.83
+2013-03-01 09:11:58.703071	52.09	0.83
+2013-03-01 09:11:58.703071	63.03	0.83
+2013-03-01 09:11:58.703071	76.13	0.83
+2013-03-01 09:11:58.703071	44.3	0.83
+2013-03-01 09:11:58.703071	44.66	0.83
+2013-03-01 09:11:58.703071	17.58	0.83
+2013-03-01 09:11:58.703071	11.32	0.83
+2013-03-01 09:11:58.703071	38.5	0.83
+2013-03-01 09:11:58.703071	82.85	0.83
+2013-03-01 09:11:58.703071	49.64	0.83
+2013-03-01 09:11:58.703071	27.23	0.83
+2013-03-01 09:11:58.703071	83.98	0.83
+2013-03-01 09:11:58.703071	31.84	0.83
+2013-03-01 09:11:58.703071	50.28	0.83
+2013-03-01 09:11:58.703071	3.73	0.83
+2013-03-01 09:11:58.703071	53.26	0.83
+2013-03-01 09:11:58.703071	29.71	0.83
+2013-03-01 09:11:58.703071	8.86	0.83
+2013-03-01 09:11:58.703071	21.01	0.83
+2013-03-01 09:11:58.703071	84.21	0.83
+2013-03-01 09:11:58.703071	19.1	0.83
+2013-03-01 09:11:58.703071	31.94	0.83
+2013-03-01 09:11:58.703071	88.93	0.83
+2013-03-01 09:11:58.703071	12.83	0.83
+2013-03-01 09:11:58.703071	29.07	0.83
+2013-03-01 09:11:58.703071	61.88	0.83
+2013-03-01 09:11:58.703071	61.41	0.83
+2013-03-01 09:11:58.703071	46.84	0.83
+2013-03-01 09:11:58.703072	95.01	62.09
+2013-03-01 09:11:58.703072	62.09	29.01
+2013-03-01 09:11:58.703072	29.01	29.01
+2013-03-01 09:11:58.703072	79.46	4.48
+2013-03-01 09:11:58.703072	4.48	4.48
+2013-03-01 09:11:58.703072	99.26	4.48
+2013-03-01 09:11:58.703072	58.77	4.48
+2013-03-01 09:11:58.703072	86.98	4.48
+2013-03-01 09:11:58.703072	30.47	4.48
+2013-03-01 09:11:58.703072	68.2	4.48
+2013-03-01 09:11:58.703072	52.44	0.79
+2013-03-01 09:11:58.703072	0.79	0.79
+2013-03-01 09:11:58.703072	25.34	0.79
+2013-03-01 09:11:58.703072	81.29	0.79
+2013-03-01 09:11:58.703072	71.68	0.79
+2013-03-01 09:11:58.703072	1.27	0.79
+2013-03-01 09:11:58.703072	56.7	0.79
+2013-03-01 09:11:58.703072	39.3	0.79
+2013-03-01 09:11:58.703072	25.91	0.79
+2013-03-01 09:11:58.703072	88.08	0.48
+2013-03-01 09:11:58.703072	0.48	0.48
+2013-03-01 09:11:58.703072	88.83	0.48
+2013-03-01 09:11:58.703072	9.0	0.48
+2013-03-01 09:11:58.703072	54.1	0.48
-- 
1.7.9.5

